### PR TITLE
CLOUDP-69682: Add global API whitelist support

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-69682

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### List
```zsh
$ mongocli iam globalWhitelist ls 
ID                         CIDR BLOCK   CREATED AT
5f3cf81b89034c6b3c0a528e   172.20.0.1   2020-08-19T16:13:08Z
```

```zsh
$ mongocli iam globalWhitelist ls -o json
{
  "results": [
    {
      "id": "5f3cf81b89034c6b3c0a528e",
      "cidrBlock": "172.20.0.1",
      "created": "2020-08-19T16:13:13Z",
      "description": "CLI",
      "type": "GLOBAL_ROLE",
      "updated": "2020-08-19T16:13:13Z"
    }
  ],
  "links": [
    {
      "rel": "self",
      "href": "http://mms:8080/api/public/v1.0/admin/whitelist?pageNum=1\u0026itemsPerPage=100"
    }
  ],
  "totalCount": 1
}
```

### Describe
```zsh
$ mongocli iam globalWhitelist describe 5f3cf81b89034c6b3c0a528e
ID                         CIDR BLOCK   CREATED AT
5f3cf81b89034c6b3c0a528e   172.20.0.1   2020-08-19T16:15:46Z
```

```zsh
$ongocli iam globalWhitelist describe 5f3cf81b89034c6b3c0a528e -o json
{
  "id": "5f3cf81b89034c6b3c0a528e",
  "cidrBlock": "172.20.0.1",
  "created": "2020-08-19T16:15:42Z",
  "description": "CLI",
  "type": "GLOBAL_ROLE",
  "updated": "2020-08-19T16:15:42Z"
}
```
### Create
```zsh
$ mongocli iam globalWhitelist create --cidr 172.20.0.2 --desc test  -P om
Global whitelist entry '5f3d5694fb9aa6294e22e494' created.
``` 

```zsh
$ mongocli iam globalWhitelist create --cidr 172.20.0.3 --desc test  -P om -o json
{
  "id": "5f3d569efb9aa6294e22e49a",
  "cidrBlock": "172.20.0.3",
  "created": "2020-08-19T16:43:10Z",
  "description": "test",
  "type": "GLOBAL_ROLE",
  "updated": "2020-08-19T16:43:10Z"
}
```

### Delete
```zsh
$ mongocli iam globalWhitelist rm 5f3d5449fb9aa6294e22e2be  -P om
? Are you sure you want to delete: 5f3d5449fb9aa6294e22e2be Yes
Whitelist entry '5f3d5449fb9aa6294e22e2be' deleted
```

### CODE REVIEW

File: internal/cli/iam/globalwhitelist/description.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package globalwhitelist
+
+const (
+	short           = "Manage the access whitelist for Global API Key."
+	createWhitelist = "Create an access whitelist for Global API Key."
+	deleteWhitelist = "Delete a database user from Global API Key."
```
Comment (andreaangiolillo) [inline internal/cli/iam/globalwhitelist/description.go:19]:
```suggestion
	deleteWhitelist = "Delete an access whitelist for Global API Key."
```

File: internal/store/global_api_keys_whitelist.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+type GlobalAPIKeyWhitelistLister interface {
+	GlobalAPIKeyWhitelists(*atlas.ListOptions) (*opsmngr.GlobalWhitelistAPIKeys, error)
+}
+
+type GlobalAPIKeyWhitelistDescriber interface {
+	GlobalAPIKeyWhitelist(string) (*opsmngr.GlobalWhitelistAPIKey, error)
+}
+
+type GlobalAPIKeyWhitelistCreator interface {
+	CreateGlobalAPIKeyWhite(*opsmngr.WhitelistAPIKeysReq) (*opsmngr.GlobalWhitelistAPIKey, error)
```
Comment (robcarlan-mlab) [inline internal/store/global_api_keys_whitelist.go:37]: [np]`CreateGlobalAPIKeyWhite` -> `CreateGlobalAPIKeyWhitelist`

File: internal/store/global_api_keys_whitelist.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+type GlobalAPIKeyWhitelistDeleter interface {
+	DeleteGlobalAPIKeyWhitelist(string) error
+}
+
+// GlobalAPIKeyWhitelists encapsulates the logic to manage different cloud providers
```
Comment (robcarlan-mlab) [inline internal/store/global_api_keys_whitelist.go:44]: [np] comments

**Review by andreaangiolillo — APPROVED** [review]
One change but it is not a blocker. LGTM! Great work! 🚀

File: internal/cli/iam/globalwhitelist/delete.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+			return opts.Prompt()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+	cmd.Flags().BoolVar(&opts.Confirm, flag.Force, false, usage.Force)
+	cmd.Flags().StringVar(&opts.apiKey, flag.APIKey, "", usage.APIKey)
+
+	_ = cmd.MarkFlagRequired(flag.Role)
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/cli/iam/globalwhitelist/delete.go]: copy/pasta

File: internal/cli/iam/globalwhitelist/description.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+	listWhitelist   = "List Atlas whitelist entries for Global API Key."
+	describeEntry   = "Return one Global Whitelist Entry using its unique identifier."
```
[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/cli/iam/globalwhitelist/description.go:21]: they could, I've have issues in other packages of clash so I won't fix this here but feel free to improve on the idea, I even think we should consider getting rid of this file

File: internal/store/global_api_keys_whitelist.go (hunk 2 above)
Comment (gssbzn) [reply internal/store/global_api_keys_whitelist.go]: fixed

File: internal/cli/iam/globalwhitelist/description.go (hunk 1 above)
Comment (gssbzn) [reply internal/cli/iam/globalwhitelist/description.go]: copy/pasta and wrong also in the original, thanks for catching this

File: internal/cli/iam/organizations/apikeys/whitelist/delete.go [hunk 6]
```diff
@@ -67,7 +67,5 @@ func DeleteBuilder() *cobra.Command {
 
 	cmd.Flags().StringVar(&opts.OrgID, flag.OrgID, "", usage.OrgID)
 
-	_ = cmd.MarkFlagRequired(flag.Role)
```
Comment (gssbzn) [inline internal/cli/iam/organizations/apikeys/whitelist/delete.go:70]: drive by fix

File: internal/cli/iam/organizations/apikeys/whitelist/description.go [hunk 7]
```diff
@@ -16,6 +16,6 @@ package whitelist
 const (
 	short           = "Manage the access whitelist for your API Key."
 	createWhitelist = "Create an access whitelist for your API Key."
-	deleteWhitelist = "Delete a database user from your API Key."
```
Comment (gssbzn) [inline internal/cli/iam/organizations/apikeys/whitelist/description.go:19]: drive by fix

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**